python/activity_stream/widget_attachment_group.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from sgtk.platform.qt import QtCore, QtGui
+import sgtk
+
+from .label_widgets import LargeAttachmentThumbnail, SmallAttachmentThumbnail
+from .data_manager import ActivityStreamDataHandler
+
+from .ui.attachment_group_widget import Ui_AttachmentGroupWidget
+
+class AttachmentGroupWidget(QtGui.QWidget):
+    """
+    Widget that acts as a container for note attachments.
+    This holds both a scaled down preview and a large size
+    representation of a set of given attachments.
+    """
+    
+    OFFSET_NONE, OFFSET_LARGE_THUMB, OFFSET_SMALL_THUMB = range(3)
+    
+    def __init__(self, parent, attachment_data):
+        """
+        Constructor
+        
+        :param parent: QT parent object
+        """
+        QtGui.QWidget.__init__(self, parent)
+        
+        # now load in the UI that was created in the UI designer
+        self.ui = Ui_AttachmentGroupWidget() 
+        self.ui.setupUi(self)
+        
+        self._bundle = sgtk.platform.current_bundle()
+        
+        self._large_thumbnails = {}
+        self._small_thumbnails = {}
+        self._other_widgets = []
+        
+        self._attachment_data = attachment_data
+        
+        self.ui.attachment_frame.setVisible(False)
+        
+        current_row = 0
+        current_col = 0
+        max_col = 0
+        
+        for data in self._attachment_data:
+            obj = SmallAttachmentThumbnail(data, self.ui.preview_frame)
+            obj.clicked.connect(self._toggle_large_thumbnails)            
+            self.ui.preview_layout.addWidget(obj, current_row, current_col)
+            self._small_thumbnails[data["id"]] = obj
+            
+            
+            if current_col > 4:
+                current_col = 0
+                current_row += 1
+            else:
+                current_col += 1
```

Comment:
it is actually using a gridlayout. Is there an easier/more elegant way to do this?